pkg/actions/testdata/lua/catalogexport_unity.lua

```diff
@@ -0,0 +1,170 @@
+local action = {
+    repository_id = "myRepo",
+    commit_id = "myCommit",
+    branch_id = "myBranch",
+}
+
+-- table names must be unique
+local test_cases = {
+    {
+        name = "failed_not_delta_type",
+        tables = {
+            ["my_table_not_delta"] = {
+                td = {
+                    type = "notDelta",
+                    catalog = "ok",
+                    name = "notDelta",
+                },
+            },
+        },
+        error = "registration failed",
+    },
+    {
+        name = "failed_no_catalog_name",
+        tables = {
+            ["my_table_no_catalog"] = {
+                td = {
+                    type = "delta",
+                    name = "noCatalog",
+                },
+            },
+        },
+        error = "catalog name is required",
+    },
+    {
+        name = "failed_no_name",
+        tables = {
+            ["my_table_no_name"] = {
+                td = {
+                    type = "delta",
+                    catalog = "ok",
+                },
+            },
+        },
+        error = "table name is required",
+    },
+    {
+        name = "failed_schema_creation",
+        tables = {
+            ["my_table_schema_failure"] = {
+                td = {
+                    type = "delta",
+                    catalog = "ok",
+                    name = "schemaFailure",
+                },
+            },
+        },
+        schema_failure = true,
+        error = "failed creating/getting catalog's schema",
+    },
+    {
+        name = "success_all_tables",
+        tables = {
+            ["my_table_success"] = {
+                status = "SUCCEEDED",
+            },
+            ["my_table2_success"] = {
+                status = "SUCCEEDED",
+            },
+            ["my_table3_success"] = {
+                status = "SUCCEEDED",
+            },
+        },
+    },
+    {
+        name = "mixed_statuses",
+        tables = {
+            ["my_table_failure"] = {
+                status = "FAILED",
+            },
+            ["my_table2_success_2"] = {
+                status = "SUCCEEDED",
+            },
+            ["my_table3_failure"] = {
+                status = "FAILED",
+            },
+        },
+    },
+}
+
+-- Loads a mock table (descriptor) extractor
+local function load_table_descriptor(tables)
+    package.loaded["lakefs/catalogexport/table_extractor"] = {
```

Comment:
love that package.loaded solution, I'll change that in glue exporter at some point for the lakeFS client!

Reply:
<img src="https://media0.giphy.com/media/ejTTWRtnrxZMt5SWLy/giphy.gif"/>